image/image.go

```diff
@@ -198,3 +200,44 @@ func createRuntimeBundle(w walker, dest, refName, rootfs string) error {
 
 	return json.NewEncoder(f).Encode(spec)
 }
+
+// Generate create an imageLayou that conforms to the OCI specification on a given path
+func Generate(dest string) (retErr error) {
+	// error out if the dest directory is not empty or failed to create the file
+	s, err := ioutil.ReadDir(dest)
+	if err != nil {
+		if os.IsNotExist(err) {
+			if err2 := os.Mkdir(dest, 0755); err2 != nil {
+				return errors.Wrap(err2, "unable to create file")
+			}
+		} else {
+			return errors.Wrap(err, "unable to open file") // err contains dest
+		}
+	}
+	if len(s) > 0 {
+		return fmt.Errorf("%s is not empty", dest)
+	}
+	defer func() {
+		// if we encounter error during generating
+		// clean up the partially-denerated destination
+		if retErr != nil {
+			if err := os.RemoveAll(dest); err != nil {
+				fmt.Printf("Error: failed to remove partially-unpacked destination %v", err)
+			}
+		}
+	}()
+
+	il := imageLayout{
+		rootDir:  dest,
+		layout:   layoutStr,
+		ref:      refTag,
+		manifest: manifestStr,
+		config:   configStr,
+		tarList: []tarContent{
+			tarContent{&tar.Header{Name: "example", Size: 7, Mode: 0600}, []byte("example")},
```

Comment:
I don't think we want to hard-code any tar content for this command.

Also, I think we want to give users a way to pick different image-layout backends (e.g. tar-backed image-layouts, directory-backed image-layouts, …).  For example, the `oci-image-init` I've floated in #5 [takes a `--type` option][1].

[1]: https://github.com/opencontainers/image-tools/pull/5/files#diff-0cf1d5acd82a85146cf1eaf17df69bbcR21